pymc/logprob/transforms.py

```diff
@@ -617,6 +617,25 @@ def measurable_special_exp_to_exp(fgraph, node):
         return [1 / (1 + pt.exp(-inp))]
 
 
+@node_rewriter([pow])
+def measurable_power_expotent_to_exp(fgraph, node):
+    """Convert power(base, x) of `MeasurableVariable`s to exp form."""
+    base, inp_exponent = node.inputs
+
+    # if base is measurable then it is power(x, exponent) and not power(base, x) as expected
+    # and should use PowerTransform
+    if check_potential_measurability([base], fgraph.preserve_rv_mappings.rv_values.keys()):
+        return None
+
+    base = CheckParameterValue("base >= 0")(base, pt.all(pt.ge(base, 0.0)))
+
+    # check whether inp_exponent is discrete
+    if inp_exponent.type.dtype.startswith("int"):
+        return None
```

Comment:
Why can't it be discrete?

IIRC correctly the discrete case was even more flexible because we could potentially support negative values in the future, but should work equally fine for the positive support

Reply:
I've misread your orignal comment as saying no discrete variables at all, but I see now it's only for the second case. I'll remove.